keynav.c

```diff
@@ -1591,26 +1720,28 @@ void handle_commands(char *commands) {
     for (i = 0; dispatch[i].command; i++) {
       /* XXX: This approach means we can't have one command be a subset of
        * another. For example, 'grid' and 'grid-foo' will fail because when you
-       * use 'grid-foo' it'll match 'grid' first. 
+       * use 'grid-foo' it'll match 'grid' first.
        * This hasn't been a problem yet...
        */
 
       /* If this command starts with a dispatch function, call it */
-      size_t cmdlen = strcspn(tok, " \t");
-      if (!strncmp(tok, dispatch[i].command, cmdlen)) {
+      size_t cmdlen = strlen(dispatch[i].command);
+      size_t tokcmdlen = strcspn(tok, " \t");
+      if (cmdlen == tokcmdlen && !strncmp(tok, dispatch[i].command, cmdlen)) {
         /* tok + len + 1 is
          * "command arg1 arg2"
-         *          ^^^^^^^^^ <-- this 
+         *          ^^^^^^^^^ <-- this
          */
         char *args = tok + cmdlen;
+
         if (*args == '\0')
           args = "";
         else
           args++;
 
         found = 1;
         dispatch[i].func(args);
-
+        break;
```

Comment:
this is ok, probably easier to read if the for loop condition is changed `dispatch[i].command && !found` instead of breaking here. (should have the same effect, though)

Reply:
Good point!  Made that change in https://github.com/mgsloan/keynav/commit/0331365944f5f5fdabc3b3e441eb6df92259bdc7